parseSchema/index.js

```diff
@@ -0,0 +1,78 @@
+"use strict";
+
+const Parse = require('parse/node');
+
+function updateSchemaOnlyWhenNecessary(existingSchemaData, intendedSchema)
+{
+  // remove any existing fields from the request.
+  Object.getOwnPropertyNames(existingSchemaData.fields).forEach(existingField => { delete intendedSchema._fields[existingField] ; });
+  return (Object.getOwnPropertyNames(intendedSchema._fields).length + Object.getOwnPropertyNames(intendedSchema._indexes).length > 0 ) ? 
+    intendedSchema.update() : Promise.resolve(existingSchemaData);
+}
+
+function setParseSchema (serverConfig) {
+
+  Parse.initialize(serverConfig.appId, null, serverConfig.masterKey);
+  Parse.serverURL = serverConfig.serverURL ;
+  
+/*
+  Notes:
+
+  (1) A Parse.Schema only represents data needed to make a request about a Parse Schema.  A better name would be SchemaRequest.
+  
+  (2) As such, fields added by Parse.Schema.prototype.addField() function and its ilks only represent intention to add/create them, and nothing else.
+      For example, the .get() method doesn't change any such "fields" in the Schema object based on the result returned from the Parse server.
+  
+  (3) Starting from Parse v2, standard Javascript Promise has replaced Parse.Promise, which unfortunately is still used in sample code
+      in ParsePlatform documentation as of this writing.
+ */
+  const contactSchema = new Parse.Schema('contacts');
```

Comment:
Are `Role` and `User` tables/classes created elsewhere?

Reply:
They are built in as far as I know.  We do need to properly populate/initialize the Role table, which is the basis for further permission setup.